### PR TITLE
signup engine no longer depends on LoginState

### PR DESCRIPTION
**Pull request by maxtaco**
Instead, we're using a `ProvisionalLoginContext` which works instead of an `Account` object. This technique allowed us through with minimal code changes, but we can certainly do some deeper refactoring.

For now, we're going to propagate the changes back into LoginState at the end of the signup, but later, once we change Login over, we can move those changes in a different object.

### CODE REVIEW

**Comment by maxtaco** [comment]
I didn't really use the `MetaContext` refactor in this patch.  🤔

File: go/engine/device_wrap.go [hunk 1]
```diff
@@ -85,36 +85,34 @@ func (e *DeviceWrap) Run(m libkb.MetaContext) error {
...
+	err = m.ActiveDevice().Set(m, e.args.Me.GetUID(), deviceID, e.signingKey, e.encryptionKey, e.args.DeviceName)
+	if err != nil {
+		return err
+	}
 
-		device := kgEng.device()
+	// Sync down secrets for future offline login attempts to work.
+	// This will largely just download what we just uploaded, but it's
+	// easy to do this way.
+	w := m.ActiveDevice().SyncSecrets(m)
```
Comment (patrickxb) [inline go/engine/device_wrap.go]: This looks a bit odd to me using `m` to get `ActiveDevice` then passing same `m` into SyncSecrets...I guess it's fine...

**Comment by patrickxb** [comment]
Does this protect against concurrent signups?  Or login and signup running concurrently?

**Comment by maxtaco** [comment]
Ok, i'll give it another whirl with the above changes...

File: go/engine/signup_join.go [hunk 2]
```diff
@@ -141,7 +141,12 @@ func (s *SignupJoinEngine) WriteOut(m libkb.MetaContext, salt []byte) error {
 		return err
 	}
 	var nilDeviceID keybase1.DeviceID
-	return lctx.SaveState(s.session, s.csrf, s.username, s.uid, nilDeviceID)
+	if err := lctx.SaveState(s.session, s.csrf, s.username, s.uid, nilDeviceID); err != nil {
+		return err
+	}
+	// Switching to a new user is an operation on the GlobalContext, and will atomically
+	// update the config file and alter the current ActiveDevice. So farm out to over there.
+	return m.SwitchUserNewConfig(s.uid, s.username, salt, nilDeviceID)
```
Comment (patrickxb) [inline go/engine/signup_join.go]: side note:  I think it's safe to remove `lastPassphrase` from `SignupJoinEngine` struct

**Pull request merged**